### ISSUE TITLE
Reading more than 200 bytes using unroll

### ISSUE THREAD

**Issue by ech0-py** [iovisor/bpftrace#804]
**App version:** latest bpftrace v0.9.1 installed via snap

```
tracepoint:syscalls:sys_enter_read,
tracepoint:syscalls:sys_enter_write
/ pid == $1 /
{
  @fd = args->fd;
  @buf_raw = args->buf;
  @size = args->count;
}


tracepoint:syscalls:sys_exit_read,
tracepoint:syscalls:sys_exit_write
/ (pid == $1) && (@size != 8) && (args->ret > 0)/
{
  printf("%s fd:%d buf:%d size:%d ret:%d\n\n", 
             probe, @fd, @buf_raw, @size, args->ret);

  $true_size = args->ret;
  $part_sz = 200;
  $parts = $true_size / $part_sz;
  $parts_remainder = $true_size % $part_sz;
  $addr = @buf_raw;

  unroll(20) {
    if ($parts > 0) {
      printf("addr: %d\n", $addr);
      @b = str($addr, $part_sz);
      // printf("%s", @b);
      clear(@b);
      $addr = $addr + $part_sz;
      $parts--;
    }
  }
  @b = str($addr, $parts_remainder);
  //printf("%s", @b);
  printf("\n\n");
}
```

Running this script via `sudo -E BPFTRACE_STRLEN=200 bpftrace af_unix.bt 16298` I'm getting the following output which means that the loop/condition is working correctly:

```
tracepoint:syscalls:sys_exit_read fd:20 buf:1489636016 size:4095 ret:603

addr: 1489636016
addr: 1489636216
addr: 1489636416
```
But when I trying to uncomment printf calls - getting the error on start:
```
error: <unknown>:0:0: in function tracepoint:syscalls:sys_exit_read i64 (i8*): Looks like the BPF stack limit of 512 bytes is exceeded. Please move large on stack variables into BPF per-cpu array map.
```
What I'm doing wrong, and is this a legal method to read the entire buffer which exceeds 200bytes?

**Comment by ex3me0** [comment]
Hmm... Seems all is OK if I'll change the
```
@b = str($addr, $part_sz);
printf("%s", @b);
```
to
```
printf("%s", str($addr, $part_sz));
```
Everything working, but still confused what happens when I use intermediate variable

**Comment by mmisono** [comment]
I think this is fixed by #1009. In short, a temporal buffer allocated for reading map contents is not freed after use.

(By the way, it seems in this example it is enough to use local variable (`$b`) instead of a map variable (`@b`).)
